### PR TITLE
feat(GatewayActivity): add missing fields

### PR DESCRIPTION
**Pull request by advaith1**
https://github.com/discord/discord-api-docs/pull/2219

### CODE REVIEW

File: v8/payloads/gateway.ts [hunk 1]
```diff
@@ -59,11 +59,13 @@ export type GatewayPresenceClientStatus = Partial<Record<'desktop' | 'mobile' |
  * https://discord.com/developers/docs/topics/gateway#activity-object-activity-structure
  */
 export interface GatewayActivity {
+	id?: string;
```
Comment (Vaporox) [inline v8/payloads/gateway.ts:62]:
```suggestion
	id: string;
```

Comment (advaith1) [reply v8/payloads/gateway.ts:62]: I marked it as optional because bots can't send it and the same interface is used for sending

Comment (Vaporox) [reply v8/payloads/gateway.ts:62]: no, that was changed with #30 :eyes:

**Review by vladfrangu — APPROVED** [review]
LGMT, waiting on Discord (debating this) to see how it progresses before merging

File: v8/payloads/gateway.ts [hunk 2]
```diff
@@ -59,22 +59,31 @@ export type GatewayPresenceClientStatus = Partial<Record<'desktop' | 'mobile' |
...
 	party?: GatewayActivityParty;
 	assets?: GatewayActivityAssets;
 	secrets?: GatewayActivitySecrets;
 	instance?: boolean;
 	flags?: ActivityFlags;
 }
 
+export enum ActivityPlatform {
+	Samsung = 'samsung',
+	Xbox = 'xbox',
```
Comment (Vaporox) [inline v8/payloads/gateway.ts:84]:
```suggestion
	Desktop = 'desktop',
	Samsung = 'samsung',
	Xbox = 'xbox',
```

File: v8/payloads/gateway.ts [hunk 3]
```diff
@@ -132,3 +143,7 @@ export enum ActivityFlags {
 	SYNC = 1 << 4,
 	PLAY = 1 << 5,
 }
+export interface GatewayActivityButton {
```
Comment (vladfrangu) [inline v8/payloads/gateway.ts:146]: New line
```suggestion

export interface GatewayActivityButton {
```

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**